x-pack/filebeat/input/awscloudwatch/input.go

```diff
@@ -120,36 +118,54 @@ func (in *cloudwatchInput) Run(inputContext v2.Context, pipeline beat.Pipeline)
 	cwPoller := newCloudwatchPoller(
 		log.Named("cloudwatch_poller"),
 		in.metrics,
-		in.awsConfig.Region,
+		region,
 		in.config)
 	logProcessor := newLogProcessor(log.Named("log_processor"), in.metrics, client, ctx)
-	cwPoller.metrics.logGroupsTotal.Add(uint64(len(logGroupNames)))
+	cwPoller.metrics.logGroupsTotal.Add(uint64(len(logGroupIDs)))
 	cwPoller.startWorkers(ctx, svc, logProcessor)
-	cwPoller.receive(ctx, logGroupNames, time.Now)
+	cwPoller.receive(ctx, logGroupIDs, time.Now)
 	return nil
 }
 
-func parseARN(logGroupARN string) (string, string, error) {
-	arnParsed, err := arn.Parse(logGroupARN)
-	if err != nil {
-		return "", "", fmt.Errorf("error Parse arn %s: %w", logGroupARN, err)
-	}
+// fromConfig is a helper to parse input configurations and derive logGroupIDs & aws region
+// Returned logGroupIDs could be empty, which require other fallback mechanisms to derive them.
+// See getLogGroupNames for example.
+func fromConfig(cfg config, awsCfg awssdk.Config) (logGroupIDs []string, region string, err error) {
+	// LogGroupARN has precedence over LogGroupName & RegionName
+	if cfg.LogGroupARN != "" {
+		parsedArn, err := arn.Parse(cfg.LogGroupARN)
+		if err != nil {
+			return nil, "", fmt.Errorf("failed to parse log group ARN: %w", err)
+		}
 
-	if strings.Contains(arnParsed.Resource, ":") {
-		resourceARNSplit := strings.Split(arnParsed.Resource, ":")
-		if len(resourceARNSplit) >= 2 && resourceARNSplit[0] == "log-group" {
-			return resourceARNSplit[1], arnParsed.Region, nil
+		if parsedArn.Region == "" {
+			return nil, "", fmt.Errorf("failed to parse log group ARN: missing region")
 		}
+
+		// refine to match AWS API parameter regex of logGroupIdentifier
+		groupId := strings.TrimSuffix(cfg.LogGroupARN, ":*")
+		logGroupIDs = append(logGroupIDs, groupId)
+
+		return logGroupIDs, parsedArn.Region, nil
 	}
-	return "", "", fmt.Errorf("cannot get log group name from log group ARN: %s", logGroupARN)
-}
 
-// getLogGroupNames uses DescribeLogGroups API to retrieve all log group names
-func getLogGroupNames(svc *cloudwatchlogs.Client, logGroupNamePrefix string, logGroupName string) ([]string, error) {
-	if logGroupNamePrefix == "" {
-		return []string{logGroupName}, nil
+	// then fallback to LogrGroupName
+	if cfg.LogGroupName != "" {
+		logGroupIDs = append(logGroupIDs, cfg.LogGroupName)
+	}
+
+	// finally derive region
+	if cfg.RegionName != "" {
+		region = cfg.RegionName
+	} else {
+		region = awsCfg.Region
 	}
 
+	return logGroupIDs, region, nil
+}
+
+// getLogGroupNames uses DescribeLogGroups API to retrieve all log group names
+func getLogGroupNames(svc *cloudwatchlogs.Client, logGroupNamePrefix string) ([]string, error) {
 	// construct DescribeLogGroupsInput
 	describeLogGroupsInput := &cloudwatchlogs.DescribeLogGroupsInput{
 		LogGroupNamePrefix: awssdk.String(logGroupNamePrefix),
```

Comment:
Might the attribute `IncludeLinkedAccounts: awssdk.Bool(true)` be missing here? For me, it seems to be required to get the log groups from (all) linked accounts. Correct me, if I'm wrong!

Reply:
@stefanhettich really good point and thank you bringing this to my attention :) 

Yes, per the API DescribeLogGroups [1] `includeLinkedAccounts` needs to be true to include linked accounts.

Further, this opens up another input possibility. If we utilize `includeLinkedAccounts = true` & use returned `LogGroupArn`, then users can utilize `logGroupNamePrefix`and still use linked accounts. 

I will validate this in my setup and update accordingly :) 


[1] https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_DescribeLogGroups.html

Reply:
@kaiyan-sheng fyi, this is something we didn't consider :)

Reply:
@stefanhettich  again, this was a great input and I am considering adding support to this. But I will add this through a dedicated PR and make sure both are included in the next release (aiming for 8.15.4).

Also please check the original issue #36642 and comment [1] to track updates 

@kaiyan-sheng  will open a follow-up as discussed :) 

[1] - https://github.com/elastic/beats/issues/36642#issuecomment-2406110947

Reply:
Awesome! Thanks a lot for taking care. Feel free to reach out, if I can help with anything! :)

Reply:
Opened dedicated PR https://github.com/elastic/beats/pull/41206 to add support to this :) Will resolve this conversation now